packages/cli/add/commands/project.js

```diff
@@ -0,0 +1,87 @@
+import prompts from 'prompts';
+import path from 'path';
+import fs from 'fs';
+import ts from 'typescript';
+
+import {projectTemplate} from '../templates.js';
+import {createDir} from '../../utils/createDir.js';
+import {createFile} from '../../utils/createFile.js';
+
+export async function addProject() {
+  const response = await prompts({
+    type: 'text',
+    name: 'projectName',
+    message: 'Project Name',
+    initial: 'myProject',
+    validate: value =>
+      isValidProjectName(value) ? 'Project already exists.' : true,
+  });
+  await createProject(response.projectName);
+}
+
+/* Is it OK to check if the project exists using this? */
+function isValidProjectName(projectName) {
+  return fs.existsSync(path.resolve('src', `${projectName}.ts`));
+}
+
+function isProjectArray(node) {
+  if (!ts.isArrayLiteralExpression(node)) return false;
+  const projectPropertyAssignment = node.parent;
+  if (
+    !(
+      projectPropertyAssignment &&
+      ts.isPropertyAssignment(projectPropertyAssignment) &&
+      ts.isIdentifier(projectPropertyAssignment.name) &&
+      projectPropertyAssignment.name.escapedText === 'project'
+    )
+  )
+    return false;
+  return true;
+}
+
+async function createProject(projectName) {
+  await createFile(path.resolve('src', `${projectName}.ts`), projectTemplate());
```

Comment:
Yes, but I think this check should be performed at the very beginning when testing if the project is a motion canvas one.
Generally, at the beginning we should figure out these three things:
1. is this a motion canvas project? (abort otherwise)
2. can we locate the config file? (store the path for later)
3. what language does this project use? (store that for later too)